server/user-groups.ts

```diff
@@ -355,4 +358,31 @@ export class GlobalAuth extends Auth {
 		this.save();
 		return true;
 	}
+	setSection(id: ID, sectionid: RoomSection, username?: string) {
+		if (!username) username = id;
+		const user = Users.get(id);
+		if (user) {
+			user.updateIdentity();
+			username = user.name;
+			Rooms.global.checkAutojoin(user);
+		}
+		if (!super.has(id)) this.set(id, ' ', username);
+		this.sectionLeaders.set(id, sectionid);
+		void this.save();
+		return this;
+	}
+	deleteSection(id: ID) {
+		if (!this.sectionLeaders.has(id)) return false;
+		this.sectionLeaders.delete(id);
+		if (super.get(id) === ' ') {
+			return this.delete(id);
+		}
+		const user = Users.get(id);
+		if (user) {
+			user.updateIdentity();
+			Rooms.global.checkAutojoin(user);
+		}
+		this.save();
+		return true;
+	}
```

Comment:
So, this is in some sense difficult to avoid, but if you set someone to trusted, then set and delete their section, they won't be trusted anymore.

I don't think it's worth trying to remember the trusted state from before the promotion somewhere. But I think the best solution here is probably to add a message about that when demoting them.

Reply:
Would you mind doing this? I won’t be home for a bit.

Reply:
Restart isn't for another half-day. I added the suggestion but you're in no hurry to commit it. I think the rest of this PR is done now.